File: examples/C/mmiecho.c
//
//  MMI echo query example
//

//  Lets us build this source without creating a library
#include "mdcliapi.c"

int main (int argc, char *argv [])
{
    int verbose = (argc > 1 && streq (argv [1], "-v"));
    mdcli_t *session = mdcli_new ("tcp://localhost:5555", verbose);

    //  This is the service we want to look up
    zmsg_t *request = zmsg_new ();
    zmsg_addstr (request, "echo");

    //  This is the service we send our request to
    zmsg_t *reply = mdcli_send (session, "mmi.service", &request);

    if (reply) {
        char *reply_code = zframe_strdup (zmsg_first (reply));
        printf ("Lookup echo service: %s\n", reply_code);
        free (reply_code);
        zmsg_destroy (&reply);
    }
    else
        printf ("E: no response from broker, make sure it's running\n");

    mdcli_destroy (&session);
    return 0;
}

File: examples/PHP/mmiecho.php
<?php
/*
 * MMI echo query example
 * 
 * @author Ian Barber <ian(dot)barber(at)gmail(dot)com>
 */
include "mdcliapi.php";

$verbose = $_SERVER['argc'] > 1 && $_SERVER['argv'][1] == '-v';
$session = new MDCli("tcp://localhost:5555", $verbose);

//  This is the service we want to look up
$request = new Zmsg();
$request->body_set("echo");

//  This is the service we send our request to
$reply = $session->send("mmi.service", $request);

if($reply) {
    $reply_code = $reply->pop();
    printf ("Lookup echo service: %s %s", $reply_code, PHP_EOL);
}